File: Difficulty: Easy/Intersection of Two Sorted Arrays/intersection-of-two-sorted-arrays.cpp
class Solution {
  public:
    vector<int> intersection(vector<int>& a, vector<int>& b) {
        // code here
        vector<int> result;

        int i = 0, j = 0;
    
        while (i < a.size() && j < b.size()) {
    
            // Skip duplicates in 'a'
            if (i > 0 && a[i] == a[i - 1]) {
                i++;
                continue;
            }
    
            // Skip duplicates in 'b'
            if (j > 0 && b[j] == b[j - 1]) {
                j++;
                continue;
            }
    
            if (a[i] == b[j]) {
                result.push_back(a[i]);
                i++;
                j++;
            }
            else if (a[i] < b[j]) {
                i++;
            }
            else {
                j++;
            }
        }
    
        return result;
    }
};

File: Difficulty: Easy/Intersection of Two Sorted Arrays/intersection-of-two-sorted-arrays.java
class Solution {
    ArrayList<Integer> intersection(int[] a, int[] b) {
        // code here
        LinkedHashSet<Integer> set=new LinkedHashSet<>();
        int index_a=0, index_b=0;
        while(index_a<a.length && index_b<b.length){
            if((a[index_a]==b[index_b]) && !set.contains(a[index_a])){
                set.add(a[index_a]);
            }else if(a[index_a]>b[index_b]){
                index_b++;
            }else{
                index_a++;
            }
        }
        return (new ArrayList<>(set));
    }
}